subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/AbsolutePathFingerprintingStrategy.java

```diff
@@ -52,18 +59,26 @@ public String normalizePath(FileSystemLocationSnapshot snapshot) {
             @Override
             public SnapshotVisitResult visitEntry(FileSystemLocationSnapshot snapshot, boolean isRoot) {
                 String absolutePath = snapshot.getAbsolutePath();
-                if (processedEntries.add(absolutePath)) {
+                if (processedEntries.add(absolutePath) && shouldFingerprint(snapshot)) {
                     builder.put(absolutePath, new DefaultFileSystemLocationFingerprint(snapshot.getAbsolutePath(), snapshot));
                 }
                 return SnapshotVisitResult.CONTINUE;
             }
-
         });
         return builder.build();
     }
 
+    private boolean shouldFingerprint(CompleteFileSystemLocationSnapshot snapshot) {
+        return !(snapshot.getType() == FileType.Directory && directorySensitivity == DirectorySensitivity.IGNORE_DIRECTORIES);
```

Comment:
Can we move this method to the `DirectorySensitivity` enum? The implementation could even be per enum constant.

Reply:
Good call.  I really like the way this worked out.  Much better now.